### ISSUE TITLE
Expand post shortcodes

### ISSUE THREAD

**Issue by cavis** [publicmediaplatform/pmp-wordpress#56]
When I put an image into the Post body, it creates a shortcode that looks something like this:

```html
[caption id=\"attachment_15\" align=\"alignnone\" width=\"300\"]<a href=\"http://wordpress.pmpdev.org/wp-content/uploads/2015/03/overly-cc9695031df31e0f69cacd9d0b5de98c26ad61f1.jpg\"><img class=\"size-medium wp-image-15\" src=\"http://wordpress.pmpdev.org/wp-content/uploads/2015/03/overly-cc9695031df31e0f69cacd9d0b5de98c26ad61f1-300x225.jpg\" alt=\"Laina Morris became the &quot;Overly Attached Girlfriend&quot; meme. She's embraced her online fame.\" width=\"300\" height=\"225\" /></a> Laina Morris became the \"Overly Attached Girlfriend\" meme. She's embraced her online fame.[/caption]
```

Same thing happens for audio files, and presumably many things you might insert into a WP post.

These shortcodes should be expanded into HTML before putting them in `contentencoded`.  And that should also prevent them from showing up in the story `description`.

Seems like maybe `do_shortcode()` would fix this.  Or would that also break the ability to roundtrip the Post body?

**Comment by rnagle** [comment]
A call to  `apply_filters('the_content', $post->post_content)` is the way WordPress deals with shortcodes (and any other modifications to post content) before rendering the post body for the front end.

Read more about [`apply_filters`](https://codex.wordpress.org/Function_Reference/apply_filters).

Fixed in 380cc4c764d4bf323937713981c5060f8be4b65b.

**Comment by cavis** [comment]
Looks good.

Enhancement: improve the round-trip-ability of Wordpress content with #58 

Upgraded to a Bug: need to remove the newlines in the contentencoded/description, as they would break a non-Wordpress CMS #53
